packages/frontend/src/utils/encryption/index.ts

```diff
@@ -21,3 +22,25 @@ export const encryptWallet = async (password: string): Promise<void> => {
         }
     }
 };
+
+export const decryptWalletKeys = async (password: string): Promise<void> => {
+    const encryptedkeyStore = new nearApiJs.keyStores.BrowserLocalStorageKeyStore(
+        new EncrytedLocalStorage(createKeyFrom(password)),
+        KEYSTORE_PREFIX
+    );
+
+    const decryptedkeyStore = new nearApiJs.keyStores.BrowserLocalStorageKeyStore(
+        window.localStorage,
+        KEYSTORE_PREFIX
+    );
+
+    const allAccounts = wallet.getAccountsLocalStorage();
+
+    for (let accountId in allAccounts) {
+        const keyPair = await encryptedkeyStore.getKey(CONFIG.NETWORK_ID, accountId) as KeyPairEd25519;
```

Comment:
Some extra checks for key validation? For example some conditions with `ED25516_TAG` like in the encryption code.

Reply:
Dont need, if key readable by keyStore, it is valid.